link_validator.py

```
@@ -0,0 +1,576 @@
+This module provides comprehensive link validation for all link types:
+- Internal page links (clean URLs)
+- Asset references (CSS, JS, images)
+- External links (social media, email)
+- API endpoints
+- Anchor links within pages
+
+Integrates with the server.py routing logic to validate that clean URLs
+properly map to existing HTML files in the pages/ directory.
+"""
+
+import os
```

Ruff (0.12.2)

`os` imported but unused

Remove unused import: `os`

(F401)